src/RevSysCalls.cc

```diff
@@ -0,0 +1,2570 @@
+ //    current pipeline is executed until completion
+ //    */
+ //    CtxSwitchAlert(ChildPID);
+
+ //    // Parent's return value is the child's PID
+ //    RegFile->SetX(feature, 10, ChildPID);
+
+ //    // Child's return value is 0
+ //    ChildCtx->GetRegFile()->SetX(feature, 10, 0);
+
+ //    // clean up ecall state
+ //    rtval = RevProc::ECALL_status_t::SUCCESS;
+ //    ECALL.bytesRead = 0;
+
+ //  } //else
+  return rtval;
+}
+
+// 436, rev_close_range(unsigned int fd, unsigned int max_fd, unsigned int flags) 
+RevProc::ECALL_status_t RevProc::ECALL_close_range(RevInst& inst){
+  output->verbose(CALL_INFO, 2, 0, "ECALL: close_range called by thread %" PRIu32 "\n", GetActiveThreadID());
+  return RevProc::ECALL_status_t::SUCCESS;
+}           
+
+// 437, rev_openat2(int dfd, const char  *filename, struct open_how *how, size_t size) 
+RevProc::ECALL_status_t RevProc::ECALL_openat2(RevInst& inst){
+  output->verbose(CALL_INFO, 2, 0, "ECALL: openat2 called by thread %" PRIu32 "\n", GetActiveThreadID());
+  return RevProc::ECALL_status_t::SUCCESS;
+}               
+
+// 438, rev_pidfd_getfd(int pidfd, int fd, unsigned int flags) 
+RevProc::ECALL_status_t RevProc::ECALL_pidfd_getfd(RevInst& inst){
+  output->verbose(CALL_INFO, 2, 0, "ECALL: pidfd_getfd called by thread %" PRIu32 "\n", GetActiveThreadID());
+  return RevProc::ECALL_status_t::SUCCESS;
+}           
+
+
+// 439, rev_faccessat2(int dfd, const char  *filename, int mode, int flags) 
+RevProc::ECALL_status_t RevProc::ECALL_faccessat2(RevInst& inst){
+  output->verbose(CALL_INFO, 2, 0, "ECALL: faccessat2 called by thread %" PRIu32 "\n", GetActiveThreadID());
+  return RevProc::ECALL_status_t::SUCCESS;
+}            
+
+
+// 440, rev_process_madvise(int pidfd, const struct iovec  *vec, size_t vlen, int behavior, unsigned int flags) 
+RevProc::ECALL_status_t RevProc::ECALL_process_madvise(RevInst& inst){
+  output->verbose(CALL_INFO, 2, 0, "ECALL: process_madvise called by thread %" PRIu32 "\n", GetActiveThreadID());
+  return RevProc::ECALL_status_t::SUCCESS;
+}
+
+// 1000, int pthread_create(pthread_t *restrict thread,
+//                          const pthread_attr_t *restrict attr,
+//                          void *(*start_routine)(void *),
+//                          void *restrict arg);
+RevProc::ECALL_status_t RevProc::ECALL_pthread_create(RevInst& inst){
+  output->verbose(CALL_INFO, 2, 0, "ECALL: pthread_create called by thread %" PRIu32 "\n", GetActiveThreadID());
+  uint64_t tidAddr = RegFile->RV64[10]; // Used to hold the new ThreadID
+  // const pthread_attr_t *restrict attr = (const pthread_attr_t *)RegFile->RV64[11];
+  uint64_t NewThreadPC = RegFile->RV64[11];
+  uint64_t ArgPtr = RegFile->RV64[12];
+  unsigned long int NewTID = GetNewThreadID();
+  std::cout << "New TID = " << NewTID << std::endl;
+  CreateThread(NewTID, NewThreadPC, (void*)ArgPtr);
+
+  mem->WriteMem(feature->GetHartToExec(), tidAddr, sizeof(unsigned long int), &NewTID, REVMEM_FLAGS(0x00));
```

Comment:
Prefer:

 `mem->WriteMem(feature->GetHartToExec(), tidAddr, sizeof(NewTID), &NewTID, REVMEM_FLAGS(0));`

(i.e., don't pass a fixed size separately from the data being passed; pass the size of the data being passed.)